pkg/types/ccipocr3/v2/common_types.go

```diff
@@ -0,0 +1,176 @@
+package ccipocr3
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+// UnknownAddress represents a raw address with an unknown encoding.
+type UnknownAddress []byte
+
+// NewUnknownAddressFromHex creates a new UnknownAddress from a hex string.
+func NewUnknownAddressFromHex(s string) (UnknownAddress, error) {
+	b, err := NewBytesFromString(s)
+	if err != nil {
+		return nil, err
+	}
+	return UnknownAddress(b), nil
+}
+
+// String returns the hex representation of the unknown address.
+func (a UnknownAddress) String() string {
+	return Bytes(a).String()
```

Comment:
Should we prefix this by something to make it clear that its unknown?

```suggestion
	return fmt.Sprintf("unkown_%s", Bytes(a).String())
```

Reply:
Should unknown addresses string representations ever be passed in somewhere that will make use of that representation (e.g like the contract reader)?

Reply:
Yeah, that would be great.

Reply:
I would rather not attach semantics to the string representation. It's mainly kept here for convenience and backwards compatibility.